src/orion/testing/state.py

```diff
@@ -179,33 +181,35 @@ def replace_file(v):
 
     def __enter__(self):
         """Load a new database state"""
-        self.singletons = update_singletons()
         self.cleanup()
         return self.init(self.make_config())
 
     def __exit__(self, exc_type, exc_val, exc_tb):
         """Cleanup database state"""
         self.cleanup()
+        orion.core.config.storage.from_dict(self.previous_config)
 
-        update_singletons(self.singletons)
-
-    def storage(self, config=None):
+    def setup_storage(self, config=None):
         """Return test storage"""
+        self.previous_config = orion.core.config.storage.to_dict()
+        orion.core.config.storage.from_dict(config)
+
         if config is None:
-            return get_storage()
+            self.storage = setup_storage()
+            return self.storage
 
         try:
+            self.storage_config = copy.deepcopy(config)
             config["of_type"] = config.pop("type")
-            db = storage_factory.create(**config)
-            self.storage_config = config
+            self.storage = storage_factory.create(**config)
         except SingletonAlreadyInstantiatedError:
```

Comment:
Does it make any sense to keep the error SingletonAlreadyInstantiatedError? I don't see why.

Reply:
done